### DISCUSSION TITLE
Feature request : allow for cwd truncation in title

### DISCUSSION THREAD

**Discussion by FantomeBeignet** [ghostty-org/ghostty#4095] [Feature Requests, Ideas] [answered]
I think it could be cool to allow the user to configure how many components of the cwd's path the window title displays, as long paths can take a lot of space and sometimes just the last one or two components are enough to know where you are on your system.

**Comment by mitchellh** [comment, marked as answer]
I think this is better solved in the shell, not the terminal. The shell can send terminal titles.

**Reply by FantomeBeignet** [reply]
I've tried changing the title using commands from [the archlinux wiki](https://wiki.archlinux.org/title/Zsh#xterm_title) but that does not do anything, am I doing something wrong/not understanding something or is this a bug in ghostty ?

**Reply by jcollie** [reply]
Ghostty just displays what the shell sends. If you're using something like starship, oh-my-posh, or powerline10k it's even a bit more complicated.

**Reply by FantomeBeignet** [reply]
It seems that one of the programs I am using is setting this title, as running the commands from a blank shell spawned with `zsh -f` sets the title correctly. However, I think my inability to track down where this is modified in my config is an argument in favor of #4096 I also opened, in addition to the ones I cite in the discussion.
